src/linear_algebra/quadratic_form.lean

```diff
@@ -436,6 +436,19 @@ begin
   rw [mul_right_comm, mul_inv_of_self, one_mul],
 end
 
+-- Change to module over rings once #6585 is merged
+lemma exists_quadratic_form_neq_zero [nontrivial M]
+  {Q : quadratic_form R₁ M} (hB₁ : Q.associated.nondegenerate) :
```

Comment:
Should we have a `Q.nondegenerate` definition to avoid having to go via `Q.associated`?

Reply:
I would incline to no; if we ever get a substantial body of lemmas with this hypothesis that could change.

Reply:
Would `Q.nondegenerate` be defined as `Q.associated.nondegenerate`?

Reply:
`Q.nondegenerate` will be useful for Sylvester's law of inertia so maybe the definition can be introduced once that comes around?

Reply:
Sounds good to me!